kaldb/src/main/java/com/slack/kaldb/zipkinApi/ZipkinService.java

```diff
@@ -92,24 +95,20 @@ protected static String convertLogWireMessageToZipkinSpan(List<LogWireMessage> m
         continue;
       }
 
-      final long messageConvertedTimestamp = convertToMicroSeconds(message.getTimestamp());
-
-      final Span span =
-          makeSpan(
-              messageTraceId,
-              Optional.ofNullable(parentId),
-              message.getId(),
-              Optional.ofNullable(name),
-              Optional.ofNullable(serviceName),
-              messageConvertedTimestamp,
-              duration,
-              messageTags);
-      String spanJson = printer.print(span);
-      traces.add(spanJson);
+      final ZipkinSpanResponse span = new ZipkinSpanResponse(message.getId(), messageTraceId);
```

Comment:
```
final ZipkinSpanResponse span = new ZipkinSpanResponse(message.getId(), messageTraceId, parentId, name,
              convertToMicroSeconds(message.getTimestamp()), null, remoteEndpoint, Math.toIntExact(duration), null, messageTags);
```

Reply:
We elected not to go with this approach as we lose the benefit of required/optional field enforcement. Today the constructor requires both traceId and id, but the rest of the fields are optional.